controllers/etcd_controller_test.go

```diff
@@ -147,654 +150,861 @@ func cmdIterator(element interface{}) string {
+							Expect(instance.Finalizers).To(BeEmpty(), "finalizer should be removed from the etcd resource")
+						})
+
+						Describe("without any statefulset", func() {
+							BeforeEach(func() {
+								mockListGetAndUpdateStatefulSet(cl, labelSelector).AnyTimes()
+							})
+
+							reconcileShouldSucceed()
+						})
+
+						Describe("with a statefulset", func() {
+							var (
+								sts *appsv1.StatefulSet
+							)
+
+							BeforeEach(func() {
+								sts = &appsv1.StatefulSet{
+									ObjectMeta: metav1.ObjectMeta{
+										Name:      instance.Name,
+										Namespace: instance.Namespace,
+									},
+								}
+								mockListGetAndUpdateStatefulSet(cl, labelSelector, sts).AnyTimes()
+							})
+
+							Describe("if statefulset is not claimed", func() {
+								reconcileShouldSucceed()
+
+								It("should not delete the statefulset", func() {
+									Expect(sts.DeletionTimestamp).To(BeNil())
+								})
+							})
+
+							Describe("if statefulset is claimed", func() {
+								BeforeEach(func() {
+									cl.EXPECT().Delete(gomock.Any(), gomock.AssignableToTypeOf(sts), gomock.Any()).DoAndReturn(
+										func(_ context.Context, _ *appsv1.StatefulSet, _ ...client.DeleteOption) error {
+											sts.DeletionTimestamp = instance.DeletionTimestamp
+											return nil
+										},
+									)
+								})
+
+								AfterEach(func() {
+									Expect(sts.DeletionTimestamp).ToNot(BeNil(), "statefulset should be deleted")
+								})
+
+								Describe("statefulset has owner references", func() {
+									BeforeEach(func() {
+										sts.OwnerReferences = []metav1.OwnerReference{
+											{
+												APIVersion:         etcdGVK.GroupVersion().String(),
+												Kind:               strings.ToLower(etcdGVK.Kind),
+												Name:               instance.Name,
+												UID:                instance.UID,
+												Controller:         pointer.BoolPtr(true),
+												BlockOwnerDeletion: pointer.BoolPtr(true),
+											},
+										}
+									})
+
+									reconcileShouldSucceed()
+								})
+
+								Describe("statefulset has owner annotations", func() {
+									BeforeEach(func() {
+										sts.Annotations = map[string]string{
+											common.GardenerOwnedBy:   client.ObjectKeyFromObject(instance).String(),
+											common.GardenerOwnerType: strings.ToLower(etcdGVK.Kind),
+										}
+									})
+
+									reconcileShouldSucceed()
+								})
+							})
+						})
+					})
+				})
+
+				Describe("create or update", func() {
+					AfterEach(func() {
+						Expect(instance.Finalizers).To(ContainElement(FinalizerName), "etcd resource should have finalizer added")
+
+						if instance.Spec.Backup.Store != nil {
+							Expect(backupSecret.Finalizers).To(ContainElement(FinalizerName), "backup secret should have finalizer added")
+						}
+					})
+
+					Describe("without services and configmaps", func() {
```

Comment:
Can you add tests for "with services and configmaps" as well to verify the Druid's update mechanism? (-> `syncServiceSpec`, `syncConfigMapData`)

Reply:
I was in two minds if I should add that case or not :-) I decided against adding it in this PR considering that that case wasn't tested in the existing test cases. I have changed the claim logic considerably (simplified it to `CreateOrUpdate` or re-create)  in implementing #186 and will cover this case as part of that PR. I hope that is ok.

Reply:
Okay sure, thank you :)